engine/src/object.cpp

```diff
@@ -5680,11 +5720,7 @@ void MCDeletedObjectsLeaveWait(bool p_dispatching)
     MCdeletedobjectpool = MCdeletedobjectpool -> parent;
     
     // The previous pool is now defunct.
-    t_pool -> defunct = true;
-    
-    // If the objectpool has no references then we can delete it.
-    if (t_pool -> references == 0)
-        MCDeletedObjectPoolDestroy(t_pool);
+	MCDeletedObjectPoolMakeDefunct(t_pool);
```

Comment:
If you could make this indent with spaces (since the rest of the handler is) that would be better.